### ISSUE TITLE
Unknown reason for crash

### ISSUE THREAD

**Issue by geffrod** [martin-tornqvist/ia#109]
Animated some weapons, fired an incinerator round at monsters some of them were righting, crash before game even registers one incendiary round was fired. 
Full video:
https://i.4cdn.org/vg/1452887135240.webm

**Comment by martin-tornqvist** [comment]
Thanks. Guess I'll go shoot some Incinerators then and hope I can recreate it.

**Comment by martin-tornqvist** [comment]
Alright, I got something. Was shooting an Incinerator at an animated weapon and a group of Deep Ones, and this assertion failed:
```
Assertion failed: !corpse_name.empty(), file actor.cpp, line 645
```
This part of the code is responsible for printing a message about a corpse being destroyed.

Looking at the monster data, I see that animated weapons are allowed to leave corpses. This seems wrong - the "weapon monster" is supposed to just disappear, and drop the weapon item instead. I think this can be fixed by just forbidding animated weapons from dropping corpses.

Again, thanks for reporting this.

**Comment by martin-tornqvist** [comment]
That was definitely the reason. It was actually easy to recreate by just animating a weapon, killing it by shooting at it, then throwing dynamite on its position.

Fixed by ae2e4d93c1c7b6eaf3ca015d7097617df62398e8
